test/run-drun/try-finally.mo

```diff
@@ -0,0 +1,264 @@
+        debugPrint "AFTER6c"
+    };
+
+    func t6d() : async () {
+        debugPrint "BEFORE6d";
+        label out try {
+            debugPrint "IN6d";
+            try {
+                debugPrint "InnerIN6d";
+                let fut = m();
+                await fut;
+                debugPrint "InnerLIVE6d";
+                await fut;
+                debugPrint "InnerLIVESTILL6d";
+                assert false;
+                debugPrint "InnerDEAD6d";
+            } finally { debugPrint "InnerOUT6d" };
+            debugPrint "DEAD6d";
+        }
+        finally { debugPrint "OUT6d" };
+        debugPrint "AFTERDEAD6d"
+    };
+
+    // `await*` tests
+    func t7() : async* () {
+        debugPrint "BEFORE7";
+        label out try {
+            debugPrint "IN7";
+            try {
+                debugPrint "InnerIN7";
+                let fut = m();
+                await fut;
+                debugPrint "InnerLIVE7";
+                await fut;
+                debugPrint "InnerLIVESTILL7";
+                assert false;
+                debugPrint "InnerDEAD7";
+            } finally { debugPrint "InnerOUT7" };
+            debugPrint "DEAD7";
+        }
+        finally { debugPrint "OUT7" };
+        debugPrint "AFTERDEAD7"
+    };
+
+    func t8() : async () {
+        try {
+            debugPrint "IN8";
+            // await* async* throw error "IN8"
+            // https://github.com/dfinity/motoko/issues/4578
+            await* async* { throw error "IN8"; () }
+        }
+        catch _ { debugPrint "CAUGHT8" }
+        finally { debugPrint "OUT8" };
+    };
+
+    func t8i() : async () {
+        // see: https://github.com/dfinity/motoko/issues/4578
+        func inner() : async* () = async* { throw error "IN8i" };
+
+        try {
+            debugPrint "IN8i";
+            await* inner()
+        }
+        catch _ { debugPrint "CAUGHT8i" }
+        finally { debugPrint "OUT8i" };
+    };
+
+    func t8t() : async () {
+        func inner() : async* () = async* { debugPrint "InnerIN8t"; await m(); assert true };
+
+        try {
+            debugPrint "IN8t";
+            await* inner()
+        }
+        finally { debugPrint "OUT8t" };
+    };
+
+    public func go() : async () {
+        // These don't trap (for the interpreters)
+        //await t1();
+        await t2();
+        ignore await t2i();
+        await t3();
+        await t4();
+        await t4f();
+        await t5();
+        await t6();
+        await t8();
+        await t8i();
+
+        // These trap, and only work on drun
+        try /*ignore*/ await t0() catch _ {};
+        try await t0r() catch _ {};
+        try await t0d() catch _ {};
+        try await t2t() catch _ {};
+        try await t6c() catch _ {};
+        try await t6t() catch _ {};
+        try await t6d() catch _ {};
+        try await t8t() catch _ {};
+        /// caveat: t7 won't return!
+        try await* t7() catch _ {} finally debugPrint "It's over";
```

Comment:
@crusso my gut feeling says `"It's over"` should be output, but it doesn't appear...

Reply:
Oh dear, that might actually be true. Did you modify the async* value translation to take an additional continuation, like the async value translation? If not, maybe that is what is missing here.

Reply:
I think `t_async` is missing a key piece. On it.